src/sundials/sundials_profiler.c

```diff
@@ -258,45 +249,39 @@ int SUNProfiler_Begin(SUNProfiler p, const char* name)
     ier   = SUNHashMap_Insert(p->map, name, (void*)timer);
     if (ier)
     {
-#ifdef SUNDIALS_DEBUG
-      slen   = strlen(name);
-      errmsg = malloc(slen * sizeof(char));
-      snprintf(errmsg,
-               128 + slen, "(((( [ERROR] in SUNProfilerBegin: SUNHashMapInsert failed with code %d while inserting %s))))\n",
-               ier, name);
-      SUNDIALS_DEBUG_PRINT(errmsg);
-      free(errmsg);
-#endif
       sunTimerStructFree(timer);
       sunStopTiming(p->overhead);
-      return (-1);
+      if (ier == -1) { return SUN_ERR_PROFILER_MAPINSERT; }
+      if (ier == -2) { return SUN_ERR_PROFILER_MAPFULL; }
```

Comment:
Should `SUNHashMap_Insert` return a `SUNErrCode`?

Reply:
I don't think so because it has unique semantics w.r.t. what the return values indicate.